students/ptfbanks/Lesson03/pf_mailroom.py

```diff
@@ -0,0 +1,93 @@
+#  MailRoom exercise lesson03 final task
+#    - Build, sort, print a table
+#--------------------------Global Region-----------------------------------#
+donations = [
+    ("Bob Williams", 5580.00),
+    ("Tom Haskell", 345.05),
+    ("Earl Jackson", 4553.45),
+    ("Les Thomas", 8934.00),
+    ("James Black", 154),
+    ("Bob Williams", 3245.50),
+    ("Tom Haskell", 3245.90),
+    ("Earl Jackson", 45.6),
+    ("Les Thomas", 325.00),
+    ("James Black", 1300),
+    ("Bob Williams", 1000),
+    ("Tom Haskell", 243.05),
+    ("Earl Jackson", 345.00),
+    ("Les Thomas", 3665.00),
+    ("James Black", 21)]
+
+#-----------------------------------Define Functions----------------------#
+#Prepare list of donors
+def get_names(list):
+    names = []
+    for entry in (list):
+        name = entry[0]
+        if name not in names:
+            names.append(name)
+    return(names)
+            
+def select_Key(table):
+    return table[1]
+
+#Create history report
+def hist_rpt(dlist):
+    names=get_names(donations)  #refresh names of donors
+    print("|   DonorName    | TotaL Given | Num Gifts   | Avrage Gift |")
+    print("|----------------|-------------|-------------|-------------|")
+    rpt=[]      #list for report rows
+    tc = 0      #set tuple member ID
+    for name in (names):
+        dsum =0      #set donor sum
+        dc = 0       #set donations count
+        tup=[]      #collector for row content
+        for entry in (dlist):    #search donations ist
+            donor = entry[0]
+            if donor == name:
+                dc=dc+1
+                dval = entry[1]
+                dsum = dsum + dval
+        davg = dsum/dc         #calculate donor average donation
+        rpt.append((name, dsum, dc, davg))  #create table entry
+    rpt.sort(key=select_Key, reverse=True)   #sort descending
```

Comment:
sorting is mentioned here (which is required reading in lesson3):
https://uwpce-pythoncert.github.io/PythonCertDevel/modules/Sequences.html#custom-sorting

for any python built-ins I highly recommend getting familiar with searching official docs:
https://docs.python.org/3.3/howto/sorting.html
https://docs.python.org/3.3/library/stdtypes.html#list.sort

and yes your implementation here is correct.